### DISCUSSION TITLE
how to save to database in optimal way?

### DISCUSSION THREAD

**Discussion by ghost** [ueberdosis/tiptap#4251] [General Questions]

**Comment by ghost** [comment]


**Reply by heymartinadams** [reply]
Yeah, it’s weird. I just signed up for the PRO plan, and am paying $149/month, and yet the ueberdosis folks are either neglecting customer support, or just haven’t hired enough people to help out.

**Reply by nperez0111** [reply]
We are a small team and this is a huge project. It is a lot of work to support. Would be happy to help with something you need @heymartinadams

**Reply by heymartinadams** [reply]
I’m sure it is, @nperez0111! But I can be forgiven to expect quicker customer support when paying $149/mo.

**Reply by heymartinadams** [reply]
Not being mean or snide, just think that at that price level customer support turnaround should be much quicker.

**Reply by nperez0111** [reply]
No worries, totally understand. Did you need anything in particular?

The OSS side can be noisy and we have a discord and slack channels for customers so priority kind of just goes towards that instead

**Reply by heymartinadams** [reply]
Gotcha. Any links to PRO-support level on Discord/Slack? Couldn’t find any in my Tiptap Cloud account.

Many issues at the moment, but the main thing is:
- outdated [NextJS template](https://github.com/ueberdosis/tiptap-templates/tree/main/templates/next-block-editor-app): used it only to discover that some of the processes aren’t used anymore; ended up costing me days of workarounds

**Reply by nperez0111** [reply]
- Discord is linked to in our docs: https://tiptap.dev/docs#community
 - Slack is for business level & enterprise support
 
re: outdated NextJS template, you made [this issue on Jul 1st](https://github.com/ueberdosis/tiptap/issues/5301) we released streaming rich-text [for AI on Jul 3rd](https://tiptap.dev/blog/release-notes/content-ai-with-more-context-and-formatted-responses) and obviously before that I was working on that feature.
I'm sorry that we haven't gotten around to updating the template, but it just hasn't hit priority for me to work on it yet. Are the docs insufficient on how to leverage it? 

If there is anything else that you need, as a pro-customer, you can also email for support at `humans@tiptap.dev`

**Reply by heymartinadams** [reply]
Thanks! I wasn’t aware of you having released the streaming feature — thanks for sharing. I’ll get back into working on Tiptap in the future once the NextJS example has been updated (prefer examples over documentation). And thanks for the email :)

**Reply by nperez0111** [reply]
If you prefer examples, all of our docs come with examples of usage & live here on this repo so you can see how it works in a "real" way.

Examples: 
https://tiptap.dev/docs/content-ai/capabilities/text-generation/manage-responses
https://tiptap.dev/docs/content-ai/capabilities/text-generation/format

**Reply by heymartinadams** [reply]
Perhaps consider hiding the NextJS template from PRO users then as well until it’s fixed, otherwise we assume it’s a fully working and usable AS-IS example.

Again, not trying to be difficult here, just pointing out the obvious. Trying to get the outdated example to work costed me several days of engineering output. Other PRO users are reporting similar issues.

https://github.com/ueberdosis/tiptap-templates/issues/72

https://github.com/ueberdosis/tiptap-templates/issues/73

**Comment by nperez0111** [comment]
To store content back to your DB, the suggestion would be to store the entire content of your editor as a JSON object (from `editor.getJSON()`) into your DB.

If bandwidth is your concern, you are either writing a book (which even then can be like 300KB) or doing premature optimization.

Theorhetically you could do some sort of a patch into the JSON with the diff of previous changes, or even send the editor transactions (which are records of changes in editor states). But this is a lot of work and would be much better handled by a debounce or a simple submit button.
